Gems/AssetValidation/Code/CMakeLists.txt

```
@@ -35,15 +35,24 @@ ly_add_target(
             Include
     BUILD_DEPENDENCIES
         PRIVATE
-            Gem::AssetValidation.Static
+            Gem::${gem_name}.Static
 )
 
+# Inject the gem name into the Module source file
+ly_add_source_properties(
+    SOURCES
+        Source/AssetValidationModule.cpp
```

Probably wouldn't work for all Gems but could possibly do
```suggestion
        Source/${gem_name}Module.cpp
```